### DISCUSSION TITLE
Feature on the connector

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#4328] [Q&A]
If those ideas seems relevant for Webots, I might try to implement at least the first one :

Could we say that a value of -1 on distanceTolerance  (resp. axisTolerance)  disable the check on angle (resp. on distance). I've some weird case where those limitation are not relevant for me... It should be easy, since it only to add "-1" to the correct range of value, and test for it in the two functions that check for angle and distance values. 

Could we publish the current TensileStrength/ShearStrength on the lock ? That could be very useful for people trying to make decision on an instrumented locking system, able to provide feedback before eventually opening if anything is done.
Considering that those values are already computed by ODE, and used on Webots to check for limit, it should be just a matter of sending them to the controller.

**Comment by omichel** [comment]
Disabling distanceTolerance with -1 would make that a connector can connect to any other connector even if very away. This is not realistic and doesn't correspond to any real device. Setting axisTolerance to &pi; disables the axis constraint, so there is no need to add a -1 value for that. Finally, returning the strength applied to a connector doesn't seem realistic to me in: I don't know any real world device doing this. I would recommend you instead to use a special supervisor or physics plug-in to implement what you need.

One of the main design principles of Webots is to model only realistic devices, e.g., devices that exist in the real world. Your proposal seems to go against this principle.

**Reply by ShuffleWire** [reply]
Okay, I understand ! 
One question however : how can I set axisTolerance to pi ? In the GUI, the best I could do is using an approximate value.
If I enter 4, webots complain and replace my value with 3.141592653589793. Does that value will be correctly interpreted as pi on next loading, considering the fact that, due to missing digit, 3.141592653589793 < pi :/

**Reply by omichel** [reply]
Yes, it should be OK. If not let us know.

**Reply by ShuffleWire** [reply]
ok I've just ran this snippet of code : (in webots code, so i use the same option (build, link etc...)
```
  double d = 0.0;
  char s[] = "3.141592653589793";
  sscanf(s, "%lf", &d);
  printf("%.20lf %.20lf %d %.20lf %d\n", 3.141592653589793, M_PI, 3.141592653589793 == M_PI, d, d == M_PI);
```
output is : 
`3.14159265358979311600 3.14159265358979311600 1 3.14159265358979311600 1`

So that look like to be okay, even if I'm not a huge fan of doing so... the == (and so < and >) return mathematically wrong value (3.14159265358979311600 is not PI) but those values seems to be "close enought" to get interpreted as the same... But on a PC with more precision, it could fail, I think...

NB : M_PI is defined as 3.14159265358979323846, which is "clearly" not the same as above. But maybe this value is chosen to ensure that with 64-bits double, we get equality with 3.141592653589793 and 3.14159265358979311600

**Reply by omichel** [reply]
Double precision is an IEEE standard. If it works on your PC, it will work everywhere with IEEE double.

**Reply by ShuffleWire** [reply]
That a solid argument ! I shall stick to it.
Meanwhile, for future reference, I've found that, https://godbolt.org/z/sK418ErMx
The same test than me, but cleaner. And the whole application seems great for testing and sharing purpose !

**Reply by ShuffleWire** [reply]
Side question, it's not the first time I'm not able to accept an answer, I can't find the button anymore... Did I missed something ?

**Reply by omichel** [reply]
That's because it was not in the "Q&A" category, but in the "Ideas" category. I changed it to "Q&A", so that you can accept the answer now.

**Reply by ShuffleWire** [reply]
Yes ! that it ! thanks ! :)
